src/parsing/utils/parsing_table.cpp

```diff
@@ -0,0 +1,133 @@
+        // first entity is a non-terminal
+        if (firsts_.count(first_entity) != 0U) {
+          for (auto &symbol : firsts_[first_entity]) {
+            if (symbol != std::string(grammar::EPSILON)) {
+              table_[productions_[prod_no].GetParent()][symbol] = std::to_string(prod_no * 100 + rule_no);
+            }
+          }
+          // if epsilon present add production under follow(A) symbols
+          if (std::find(firsts_[first_entity].begin(), firsts_[first_entity].end(), std::string(grammar::EPSILON)) !=
+              firsts_[first_entity].end()) {
+            if (follows_.count(productions_[prod_no].GetParent()) != 0U) {
+              for (auto &symbol : follows_[productions_[prod_no].GetParent()]) {
+                table_[productions_[prod_no].GetParent()][symbol] = std::to_string(prod_no * 100 + rule_no);
+              }
+            }
+          }
+        }
+      } else if (first_entity == std::string(grammar::EPSILON)) {
+        // first entity is epsilon
+        if (follows_.count(productions_[prod_no].GetParent()) != 0U) {
+          for (auto &symbol : follows_[productions_[prod_no].GetParent()]) {
+            table_[productions_[prod_no].GetParent()][symbol] = std::to_string(prod_no * 100 + rule_no);
+          }
+        }
+      }
+    }
+  }
+}
+
+std::pair<int, int> ParsingTable::GetEntry(const std::string &non_terminal_, const std::string &terminal_) {
+  std::string entry = table_[non_terminal_][terminal_];
+  int value;
+  std::stringstream ss(entry);
+  ss >> value;
+  return std::make_pair(value / 100, value % 100);
+}
+
+void ParsingTable::PrettyPrintFirsts() {
+  std::cout << "\nFIRSTS\n\n";
+  std::cout << "SYMBOL\tFIRST_SET\n";
+  for (auto &entry : firsts_) {
+    std::cout << entry.first << "\t";
+    std::cout << "{ ";
+    for (auto &v : entry.second) {
+      std::cout << v << " , ";
+    }
+    std::cout << " }\n";
+  }
+}
+
+void ParsingTable::PrettyPrintFollows() {
+  std::cout << "\nFOLLOWS\n\n";
+
+  std::cout << "SYMBOL\tFOLLOW_SET\n";
+  for (auto &entry : follows_) {
+    std::cout << entry.first << "\t";
+    std::cout << "{ ";
+    for (auto &v : entry.second) {
+      std::cout << v << " , ";
+    }
+    std::cout << " }\n";
+  }
+}
+
+void ParsingTable::PrettyPrintTable() {
+  std::cout << "\nLL(1) PARSING TABLE\n\n";
+  for (int i = 0; i <= static_cast<int>(non_terminals_.size()); i++) {
+    for (int j = 0; j <= static_cast<int>(terminals_.size()); j++) {
+      if ((i == 0) && (j == 0)) {
+        std::cout << "\t";
+      } else if (i == 0) {
+        std::cout << terminals_[j - 1] << "\t";
+      } else if (j == 0) {
+        std::cout << non_terminals_[i - 1] << "\t";
+      } else {
+        std::cout << table_[non_terminals_[i - 1]][terminals_[j - 1]] << "\t";
+      }
+      if (j == static_cast<int>(terminals_.size())) {
+        std::cout << "\n";
+      }
+    }
+  }
+}
+
+void ParsingTable::SetFirsts(utils::SymbolsMap firsts) { firsts_ = std::move(firsts); }
```

Comment:
move getters and setters to .h file
Single line functions are allowed in .h
Just to maintain uniformity with other classes.